satellite/metainfo/metainfo.go

```diff
@@ -511,47 +514,65 @@ func CreatePath(ctx context.Context, projectID uuid.UUID, segmentIndex int64, bu
 func (endpoint *Endpoint) SetAttribution(ctx context.Context, req *pb.SetAttributionRequest) (_ *pb.SetAttributionResponse, err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	_, err = endpoint.checkBucketPointers(ctx, req)
+	// try to add an attribution that doesn't exist
+	partnerID, err := bytesToUUID(req.GetPartnerId())
 	if err != nil {
-		// TODO: return correct status code for GRPC
-		endpoint.log.Sugar().Debug("related bucket id already attributed \n")
-		return &pb.SetAttributionResponse{}, err
+		return nil, Error.Wrap(err)
 	}
 
-	// TODO: add valueattribution DB access functions added in new PR.
-
-	return &pb.SetAttributionResponse{}, nil
-}
-
-// checks if bucket has any pointers(entries)
-func (endpoint *Endpoint) checkBucketPointers(ctx context.Context, req *pb.SetAttributionRequest) (resp bool, err error) {
-	//TODO: Logic of checking if bucket exists will be added in new PR.
-	// write into value attribution DB only if bucket exists but no segments or no bucket and no segments exits
-	defer mon.Task()(&ctx)(&err)
-
 	keyInfo, err := endpoint.validateAuth(ctx, macaroon.Action{
 		Op:            macaroon.ActionList,
 		Bucket:        req.BucketName,
 		EncryptedPath: []byte(""),
 		Time:          time.Now(),
 	})
 	if err != nil {
-		return false, status.Errorf(codes.Unauthenticated, err.Error())
+		return nil, status.Errorf(codes.Unauthenticated, err.Error())
+	}
+
+	// check if attribution is set for given bucket
+	_, err = endpoint.partnerinfo.Get(ctx, keyInfo.ProjectID, req.GetBucketName())
+	if err == nil {
+		return nil, Error.New("Bucket(%s) , PartnerID(%s) cannot be attributed", string(req.BucketName), string(req.PartnerId))
+	}
+
+	if !attribution.ErrBucketNotAttributed.Has(err) {
```

Comment:
add a comment, something like:
```
// try only to set the attribution, when it's missing
```

since the statement itself is a double negation, which is slightly harder to understand.

Reply:
felt that was pretty self explanatory. Will modify as suggested